istioctl/pkg/install/pre-check.go

```diff
@@ -76,16 +77,16 @@ func installPreCheck(istioNamespaceFlag string, restClientGetter genericclioptio
 	fmt.Fprintf(writer, "\n")
 	fmt.Fprintf(writer, "#2. Kubernetes-version\n")
 	fmt.Fprintf(writer, "-----------------------\n")
-	res, err := checkKubernetesVersion(v)
+	res, err := checkKubernetesVersionValid(v)
 	if err != nil {
 		errs = multierror.Append(errs, err)
 		fmt.Fprint(writer, err)
-	} else if !res {
-		msg := fmt.Sprintf("The Kubernetes API version: %v is lower than the minimum version: "+minK8SVersion, v)
+	} else if res {
+		fmt.Fprintf(writer, "Istio is compatible with Kubernetes: %v.\n", v)
+	} else {
+		msg := fmt.Sprintf("The Kubernetes API version is %v. Istio has been validated with Kubernetes API versions %v through %v.", v, minK8sVersion, maxK8sVersion)
```

Comment:
This is just a warnign, right? Not a failure?

Reply:
@howardjohn - here is the output from running the precheck with out of range versions.

```
sdake@beast-01:~/go/src/istio.io/istio$ !out
out/linux_amd64/istioctl verify-install

Checking the cluster to make sure it is ready for Istio installation...

#1. Kubernetes-api
-----------------------
Can initialize the Kubernetes client.
Can query the Kubernetes API Server.

#2. Kubernetes-version
-----------------------
The Kubernetes API version is v1.17.4. Istio has been validated with Kubernetes API versions 1.18 through 1.19.

#3. Istio-existence
-----------------------
Istio will be installed in the istio-system namespace.

#4. Kubernetes-setup
-----------------------
Can create necessary Kubernetes configurations: Namespace,ClusterRole,ClusterRoleBinding,CustomResourceDefinition,Role,ServiceAccount,Service,Deployments,ConfigMap. 

#5. SideCar-Injector
-----------------------
This Kubernetes cluster supports automatic sidecar injection. To enable automatic sidecar injection see https://istio.io/docs/setup/kubernetes/additional-setup/sidecar-injection/#deploying-an-app

-----------------------

Error: 1 error occurred:
	* The Kubernetes API version is v1.17.4. Istio has been validated with Kubernetes API versions 1.18 through 1.19.
```

I think we can soften the error to a warning as is done in Kubeadm when the version of the distro is unknown or not part of the test matrix supported by kubeadm upstream. This PR was mostly to get a conversation started about the need to warn operators about possible test matrix problems, rather than enforce specific pins.